src/python/gudhi/simplex_tree.py

```diff
@@ -397,25 +397,69 @@ def flag_persistence_generators(
+        new flag complex (with the given maximal dimension). The new complex will be smaller but with same persistent
+        homology as the original one (at least in the existing dimensions: all cycle classes with higher or equal
+        dimension than `max_expansion_dim` will be ignored even if they existed before).
 
-        A natural application is to get a simplex tree of dimension 1 from :class:`~gudhi.RipsComplex`,
-        then collapse edges, perform :meth:`expansion()` and finally compute persistence
-        (cf. :download:`rips_complex_edge_collapse_example.py <../example/rips_complex_edge_collapse_example.py>`).
+        If `max_expansion_dim` is set to `1` or less, the method is equivalent to the deprecated method
+        :meth:`collapse_edges()`.
 
-        :param nb_iterations: The number of edge collapse iterations to perform. Default is 1.
+        :param nb_iterations: The number of edge collapse iterations to perform. If `nb_iterations` is strictly less
+            than 1, the method does nothing and returns `self` (independently of the value of `inplace`). Default is 1.
         :type nb_iterations: int
+        :param max_expansion_dim: The maximal dimension to which the new complex has to be expended to. If `None`, the
+            current dimension is chosen. If `1` or less, no expansion is done and only the 1-skeleton remains.
+            Note that the final dimension of the new complex can be smaller if no higher-dimensional simplex can exist.
+            Default is `None`.
+        :type max_expansion_dim: int
+        :param inplace: If `True`, the collapse is done on this simplex tree. Otherwise, the collapse is done on a new
+            tree which is then returned. Default is `True`.
+        :type inplace: bool
+        :returns: `self` (after modifications from collapses) if `inplace` is set to `True` and a new tree resulting
+            from the collapses otherwise.
+        :rtype: SimplexTree
 
         .. warning::
-            The current simplex tree is assumed to be a graph, that is of maximal dimension 1.
-            If it is not the case, all higher dimensional simplices will get lost during the
-            reduction process and not be reinserted. To regain them, call `expansion(max_dim)` afterwards.
+            The current simplex tree is assumed to be a flag complex. If it is not the case, some information may be
+            lost during the re-expansion, i.e., the persistent homology may differ from before.
         """
         if nb_iterations < 1:
-            return
-        super()._collapse_edges(nb_iterations)
-        return self
+            return self
+        if max_expansion_dim is None:
+            max_expansion_dim = self.dimension()
+        if inplace:
+            super()._collapse_edges_inplace(nb_iterations)
+            if max_expansion_dim > 1:
+                super().expansion(max_expansion_dim)
```

Comment:
Do we need `super()` all over the place? I thought it was only needed when the current class has a method with the same name as a method from the parent, and you want to call the second one.